pages/reference/OS/configuration.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Configuration
+excerpt: Instructions for editing and customizing config.json for modifying the behavior of {{ $names.os.lower }}
+---
+
+# Configuring {{ $names.os.lower }}
+
+The behavior of {{ $names.os.lower }} can be configured by editing the `config.json` file that resides in the boot partition of the device. This file accepts a [range of fields](#valid-fields) to modify the behavior of the host OS.
+
+__Note:__ The `config.json` file is different from the [`config.txt`](https://www.balena.io/docs/reference/OS/advanced/#configtxt) file also located in the boot partition, which is used by the Raspberry Pi to set device configuration options.
+
+## Using config.json to provision a device
+
+The `config.json` file is located in the root of the `resin-boot` partition, and you may edit this file by mounting a flashed SD card and editing the file directly. Editing this file will only work if you edit before the device's first boot. Once the device has been provisioned, edits to this file have no effect as the contents of the file are copied to the device boot partition at `/mnt/boot/config.json`.
+
+## Sample config.json
+
+The following example provides all available configuration options available in `config.json`. Full details about each value may be found in the [valid fields](#valid-fields) section.
+
+```json
+{
+  "hostname": "my-custom-hostname",
+  "persistentLogging": true,
+  "country": "GB",
+  "ntpServers": "ntp-wwv.nist.gov resinio.pool.ntp.org",
+  "dnsServers": "208.67.222.222 8.8.8.8",
+  "os": {
+    "network": {
+      "connectivity": {
+        "uri": "https://api.balena-cloud.com/connectivity-check",
+        "interval": "300",
+        "response": "optional value in the response"
+      },
+      "wifi": {
+        "randomMacAddressScan": false
+      }
+    },
+    "udevRules": {
+      "56": "ENV{ID_FS_LABEL_ENC}==\"resin-root*\", IMPORT{program}=\"resin_update_state_probe $devnode\", SYMLINK+=\"disk/by-state/$env{RESIN_UPDATE_STATE}\"",
+      "64": "ACTION!=\"add|change\", GOTO=\"modeswitch_rules_end\"\nKERNEL==\"ttyACM*\", ATTRS{idVendor}==\"1546\", ATTRS{idProduct}==\"1146\", TAG+=\"systemd\", ENV{SYSTEMD_WANTS}=\"u-blox-switch@'%E{DEVNAME}'.service\"\nLBEL=\"modeswitch_rules_end\"\n"
+    },
+    "sshKeys": [
+      "ssh-rsa AAAAB3Nza...M2JB balena@macbook-pro",
+      "ssh-rsa AAAAB3Nza...nFTQ balena@zenbook"
+    ]
+  }
+}
+```
+
+## Valid fields
+
+The following fields can be used to modify the behavior of the host OS.
+
+### hostname
+
+(string) The configured hostname of the device, otherwise the device UUID is used.
+
+### persistentLogging
+
+(boolean) Enable or disable persistent logging on the device. Defaults to false.
+
+### country
+
+(string) The country in which the device is operating. Used for setting the WiFi regulatory domain.
+
+### ntpServers
+
+(string) A space-separated list of NTP servers to use for time synchronization. Defaults to `resinio.pool.ntp.org` servers.
```

Comment:
The [Network Requirements](https://www.balena.io/docs/reference/OS/network/2.x/#network-requirements) docs have a full list of NTP servers:

> NTP / UDP packets (port 123) are exchanged with:
> 0.resinio.pool.ntp.org
> 1.resinio.pool.ntp.org
> 2.resinio.pool.ntp.org
> 3.resinio.pool.ntp.org

But the Network Requirements docs fail to mention that the servers could be modified via `config.json`. So perhaps we could list the 4 servers above here, and modify the Network Requirements docs to point here for a list of servers.